File: backend/src/main/java/definition/Project.java
package definition;

public class Project {
    private String name;
    private int port;
    private String path;
    private Action[] actions;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getPort() {
        return port;
    }

    public void setPort(int port) {
        this.port = port;
    }

    public Action[] getActions() {
        return actions;
    }

    public void setActions(Action[] actions) {
        this.actions = actions;
    }

    public String getPath() {
        return path;
    }

    public void setPath(String path) {
        this.path = path;
    }
}

File: frontend/s-mock/src/definition/project.ts
import { Action } from './action';
import { Guid } from 'guid-typescript';

export class Project {
  private actions: Action[];
  private guid: string;
  private name: string;
  private port: number;
  private path: string;

  constructor() {
    this.guid = Guid.raw();
    this.name = 'Project';
    this.port = 8080;
    this.path = '/'

    this.actions = [];
  }
}
